packages/imt.sol/test/LazyIMT.ts

```diff
@@ -289,6 +305,47 @@ describe("LazyIMT", () => {
         })
     })
 
+    describe("# merkleProof", () => {
+        it("Should produce valid Merke proofs for different trees", async () => {
+            // Test different depths (key) and leafs (values)
+            const tests: { [key: number]: number[] } = {
+                1: [1],
+                2: [1, 2],
+                5: [6, 7, 8, 15, 16],
+                7: [7, 127],
+                20: [9, 14, 15, 16, 18, 26, 27, 28, 40, 128, 129]
+            }
+
+            // For each depth
+            // eslint-disable-next-line guard-for-in
+            for (const depth in tests) {
+                // For each amount of leafs
+                for (const numLeaf of tests[depth]) {
+                    // Create the tree
+                    await lazyIMTTest.init(depth)
```

Comment:
The `init` function cannot be called multiple times in a single test. The elements mapping will include entries from the previous tree. To safely do this you should take a [snapshot](https://hardhat.org/hardhat-network/docs/reference#evm_snapshot) before `init` then revert to the snapshot after executing the logic in the loop.

Reply:
done, using `evm_spapshoot` and `evm_revert`: 9e27c98ef556e3bf4c30e7ae716d126f274cd620

Reply:
[Here](https://github.com/privacy-scaling-explorations/zk-kit/commit/9e27c98ef556e3bf4c30e7ae716d126f274cd620#diff-3545c4e73324d45a6e7ca34c851fb8d726dc945310c2412c7890808ef75ba6e4R324) you're snapshot/reverting outside of the `for` loop which makes it a noop. Each test gets it's own contract instance to work with, which is why we don't need snapshots in other tests. In this test we're initing inside a loop, so the snapshot/revert should happen inside that same loop.

Reply:
far finger error, fixed: 19c0b431ff44a779a46591d8d5516083214a8319

Reply:
I think you should create a new snapshot inside the loop as well. This seems to work but [the docs](https://hardhat.org/hardhat-network/docs/reference#evm_snapshot) say a single snapshot should not be reverted to more than once.

Reply:
done: fbf2c9c4e7d3ca114d314b26c52786abd99cec6e